woocommerce/woocommerce-ios
# Product description AI: analytics for the first release

Part of #9465 



## Description


This PR enables the product description AI feature v1 for the upcoming release since the CTA is not very obvious for us to get user feedback, after confirming with product pecCkj-yl-p2#comment-277. I added a few Tracks events that I came up on my own, please feel free to suggest anything:

Event | Properties | Triggers
-- | -- | --
product_description_ai_button_tapped | source: aztec_editor | When the user taps on the button to start the product description AI flow.
product_description_ai_generate_button_tapped | is_retry: bool | When the user taps on the button to generate a product description with AI.
product_description_ai_apply_button_tapped |   | When the user taps on the button to apply the AI-generated product description to the product.
product_description_ai_copy_button_tapped |   | When the user taps on the button to copy the AI-generated product description.
product_description_ai_pause_button_tapped |   | When the user taps on the button to pause the product description generation with AI.
product_description_ai_generation_success |   | When the product description AI generation succeeds.
product_description_ai_generation_failed | error | When the product description AI generation fails.


## Testing instructions


- Log in to a WPCOM store if needed
- Go to the Products tab
- Create a product if needed
- Tap on an editable product
- Tap on the product description row
- Tap on the magic wand CTA above the keyboard --> `product_description_ai_button_tapped` should be tracked with `source: aztec_editor`
- Enter or update the product name
- Enter or update the product features
- Tap `Generate` --> `product_description_ai_generate_button_tapped` should be tracked with `is_retry: false`. when the result comes back successfully, `product_description_ai_generation_success` should be tracked
- Tap `Copy` and try pasting the content somewhere in or outside of the app --> `product_description_ai_copy_button_tapped` should be tracked
- Tap the retry icon in the middle --> the icon should become a pause button. `product_description_ai_generate_button_tapped` should be tracked with `is_retry: true`
- Tap the retry icon again, and immediately tap again  --> the icon should go back to the retry icon, and `product_description_ai_pause_button_tapped` should be tracked
- Tap `Apply` --> the product description should be replaced with the generated text, and `product_description_ai_apply_button_tapped` should be tracked

---
- [x] @jaclync tests `product_description_ai_generation_failed` when generating a product description offline 


---
